code/game/machinery/buttons.dm

```diff
@@ -101,7 +98,10 @@
 		update_icon()
 		return
 
-	return src.attack_hand(user)
+	if(user.a_intent != "harm" && !(W.flags & NOBLUDGEON))
+		return src.attack_hand(user)

```

Comment:
!NOBLUDGEON is here so clicking the button with a NOBLUDGEON item actually calls the item's afterattack. e.g. detective scanner must be able to scan the button for prints.